docker/build_scripts/manylinux-python.py

```diff
@@ -0,0 +1,168 @@
+    if python_tag.startswith(("cp", "pp")):
+        return python_tag[:2], int(python_tag[2]), int(python_tag[3:])
+    if python_tag.startswith("graalpy"):
+        return python_tag[:7], int(python_tag[7]), int(python_tag[8:])
+    raise LookupError(tag)
+
+
+@cache
+def get_all_tags(no_check: bool = False):
+    all_tags_ = set(p.name for p in INSTALL_DIR.iterdir() if p.is_dir())
+    if POLICY.startswith("manylinux"):
+        all_tags_ |= set(tag for tag in PYTHON_TAGS if ARCH in PYTHON_TAGS[tag])
+    if no_check:
+        all_tags_ |= set(PYTHON_TAGS.keys())
+    all_tags = list(all_tags_)
+    all_tags.sort(key=lambda tag: sort_key(tag))
+    return all_tags
+
+
+def add_parser_list(subparsers):
+    parser = subparsers.add_parser("list")
+    parser.set_defaults(func=_list)
+    parser.add_argument("-v", "--verbose", default=False, action="store_true")
+    parser.add_argument("-i", "--installed", default=False, action="store_true")
+
+
+def get_info_from_path(path: Path):
+    python = path / "bin" / "python"
+    script = """
+import json
+import sys
+pre_map = {"alpha": "a", "beta": "b", "candidate": "rc"}
+pv = sys.version_info
+pv_pre = pre_map.get(pv[3], "")
+if pv_pre:
+    pv_pre += str(pv[4])
+iv = sys.implementation.version
+iv_pre = pre_map.get(iv[3], "")
+if iv_pre:
+    iv_pre += str(iv[4])
+info = {
+    "pv": ".".join(str(p) for p in pv[:3]) + pv_pre,
+    "i": sys.implementation.name,
+    "iv": ".".join(str(p) for p in iv[:3]) + iv_pre,
+}
+print(json.dumps(info))
+    """
+    output = subprocess.run(
+        [str(python), "-c", script],
+        check=True,
+        text=True,
+        stdout=subprocess.PIPE,
+    ).stdout
+    return json.loads(output)
+
+
+def get_info_from_tag(tag):
+    python_tag, _ = tag.split("-")
+    if python_tag.startswith("pp"):
+        return {
+            "pv": f"{python_tag[2]}.{python_tag[3:]}",
+            "i": "pypy",
+            "iv": PYTHON_TAGS[tag][ARCH]["version"]
+        }
+    if python_tag.startswith("graalpy"):
+        return {
+            "pv": f"{python_tag[7]}.{python_tag[8:]}",
+            "i": "graalpy",
+            "iv": PYTHON_TAGS[tag][ARCH]["version"]
+        }
+    raise LookupError(tag)
+
+
+def _list(args):
+    tags = get_all_tags()
+    if args.installed:
+        tags = [tag for tag in tags if INSTALL_DIR.joinpath(tag).exists()]
+    for tag in tags:
+        print(f"{tag}{':' if args.verbose else ''}")
+        if args.verbose:
+            install_path = INSTALL_DIR.joinpath(tag)
+            installed = install_path.exists()
+            if installed:
+                info = get_info_from_path(install_path)
+            else:
+                info = get_info_from_tag(tag)
+            print(f"  installed:            {'yes' if installed else 'no'}")
+            print(f"  python version:       {info['pv']}")
+            print(f"  implemention:         {info['i']}")
+            print(f"  implemention version: {info['iv']}")
```

Comment:
Here's an implementation that supports JSON output:

```suggestion
def _list(args):
    tags = get_all_tags()
    if args.installed:
        tags = [tag for tag in tags if INSTALL_DIR.joinpath(tag).exists()]

    tag_infos = []

    for tag in tags:
        install_path = INSTALL_DIR.joinpath(tag)
        installed = install_path.exists()
        if installed:
            info = get_info_from_path(install_path)
        else:
            info = get_info_from_tag(tag)

        tag_info = {
            "identifier": tag,
            "installed": installed,
            "python_version": info["pv"],
            "implementation": info["i"],
            "implementation_version": info["iv"],
            "install_path": install_path,
        }
        tag_infos.append(tag_info)

    if args.format == 'text':
        for tag_info in tag_infos:
            if not args.verbose:
                print(tag_info["identifier"])
            else:
                print(f"{tag_info['identifier']}:")
                print(f"  installed:              {'yes' if tag_info['installed'] else 'no'}")
                print(f"  python version:         {tag_info['python_version']}")
                print(f"  implementation:         {tag_info['implementation']}")
                print(f"  implementation version: {tag_info['implementation_version']}")
    elif args.format == 'json':
        json.dump(tag_infos, sys.stdout, indent=2)
    else:
        raise ValueError(f'unknown format {args.format}')
```

It would also need 

```python
    parser.add_argument("--format", choices=["text", "json"], default="text")
```

in `add_parser_list()`.